src/coordination/coordinator_instance.cpp

```diff
@@ -166,8 +100,36 @@ auto CoordinatorInstance::TryFailover() -> void {
     return;
   }
 
-  // TODO: Smarter choice
-  auto new_main = ranges::begin(alive_replicas);
+  // for each DB in instance we get one DatabaseHistory
+  using DatabaseHistories = replication_coordination_glue::DatabaseHistories;
+  std::vector<std::pair<std::string, DatabaseHistories>> instance_database_histories;
+
+  bool success{true};
```

Comment:
Did you manage to make use of any_of or something to avoid manual dealing with success? Views are lazy so I am sure we can do something

Reply:
I did manage, but it didn't look easier to read in my opinion:
```
// for each DB in instance we get one DatabaseHistory
    using DatabaseHistories = replication_coordination_glue::DatabaseHistories;

    using GetInstanceRetType = decltype(std::declval<CoordinatorClient>().SendGetInstanceTimestampsRpc());
    using MaybeInstanceValuePairType = std::pair<std::string, GetInstanceRetType>;
    auto maybe_db_histories = ranges::views::transform(alive_replicas, [](ReplicationInstance &replica) {
                                return std::make_pair<std::string, GetInstanceRetType>(replica.InstanceName(), replica.GetClient().SendGetInstanceTimestampsRpc());
                              }) | ranges::to<std::vector<MaybeInstanceValuePairType>>();


    if (std::ranges::any_of(maybe_db_histories, [](const auto &maybe_history) {
          return std::get<1>(maybe_history).HasError();
        })) {
      spdlog::error("Aborting failover as at least one instance didn't provide per database history.");
      return;
    }

    using InstanceValuePairType = std::pair<std::string, DatabaseHistories>;
    std::vector<InstanceValuePairType> instance_database_histories;

    std::ranges::transform(maybe_db_histories, std::back_inserter(instance_database_histories), [](auto &value){
      return std::pair<std::string, DatabaseHistories>(std::get<0>(value), std::get<1>(value).GetValue()); 
    });
```

Reply:
I would argue that it is better because you don't have to think about how success variable is being changed etc. But leave it, I can change that later if needed